From: Andrea Bolognani <abologna@redhat.com>
Date: Wed, 3 Nov 2021 10:57:41 +0100
Subject: meson: Enable ZFS storage backend even more often

Commit 73a2ff36163e already improved the situation a lot. This
pushes things even further.

If the user or, more likely, the distro packager explicitly
asked for ZFS support to be enabled, then we should comply with
that request regardless of whether the necessary programs are
available at build time.

This is particularly important in the context of Debian, where
ZFS cannot be a build dependency of libvirt due to licensing
issues but it can still be an optional runtime dependency.

Signed-off-by: Andrea Bolognani <abologna@redhat.com>
Reviewed-by: Tim Wiederhake <twiederh@redhat.com>
(cherry picked from commit ca4dc9613fb3be1012ef458d62cc5e406b8e1bf3)
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index ad41983..2dce8a1 100644
--- a/meson.build
+++ b/meson.build
@@ -1950,7 +1950,7 @@ if conf.has('WITH_LIBVIRTD')
     foreach name : [ 'zfs', 'zpool' ]
       set_variable(
         '@0@_prog'.format(name),
-        find_program(name, required: get_option('storage_zfs'), dirs: libvirt_sbin_path)
+        find_program(name, required: false, dirs: libvirt_sbin_path)
       )
     endforeach
 
